app/src/androidTest/java/com/saucelabs/mydemoapp/android/view/activities/DashboardToCheckout.java

```diff
@@ -0,0 +1,211 @@
+package com.saucelabs.mydemoapp.android.view.activities;
+
+import android.view.View;
+import android.widget.HorizontalScrollView;
+import android.widget.ScrollView;
+
+import androidx.core.widget.NestedScrollView;
+import androidx.lifecycle.Lifecycle;
+import androidx.test.espresso.Espresso;
+import androidx.test.espresso.UiController;
+import androidx.test.espresso.ViewAction;
+import androidx.test.espresso.action.ScrollToAction;
+import androidx.test.espresso.action.ViewActions;
+import androidx.test.espresso.contrib.RecyclerViewActions;
+import androidx.test.espresso.matcher.ViewMatchers;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.LargeTest;
+
+import com.saucelabs.mydemoapp.android.R;
+
+import org.hamcrest.Matcher;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.os.SystemClock.sleep;
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.scrollTo;
+import static androidx.test.espresso.action.ViewActions.swipeDown;
+import static androidx.test.espresso.action.ViewActions.swipeUp;
+import static androidx.test.espresso.action.ViewActions.typeText;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.isAssignableFrom;
+import static androidx.test.espresso.matcher.ViewMatchers.isDescendantOfA;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.core.AllOf.allOf;
+import static org.hamcrest.core.AnyOf.anyOf;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class DashboardToCheckout {
+
+    @Rule
+    public ActivityScenarioRule<MainActivity> activityRule
+            = new ActivityScenarioRule<>(MainActivity.class);
+
+    @Before
+    public void setUp(){
+        activityRule.getScenario().moveToState(Lifecycle.State.RESUMED);
+    }
+
+    @Test
+    public void dashboard_Product_Test(){
+
+        //Note::Sleep Only Use For Testing Purpose Wait Screen
+
+        //Check RecyclerView
+        onView(withId(R.id.productRV)).check(matches(isDisplayed()));
+
+        sleep(1000);
```

Comment:
You mention this

` //Note::Sleep Only Use For Testing Purpose Wait Screen`

but there are multiple sleeps in this test. This is normally not a good practice, if we need to wait for animations to be done / elements to appear then please use a method that verifies if the animation is done / element is there instead of using hard sleeps

Reply:
I removed all sleeps except the first one which only waits for the splash.